Modules/Sources/PointOfSale/Controllers/PointOfSaleObservableItemsController.swift

```diff
@@ -108,12 +149,21 @@ private extension PointOfSaleObservableItemsController {
             return .initial
         }
 
-        // Loading state - preserve existing items
-        if dataSource.isLoadingProducts {
+        // Loading state - preserve existing items (both for data loading and refresh)
+        if dataSource.isLoadingProducts || refreshState == .loading {
```

Comment:
```suggestion
        // Loading state - preserve existing items (only use for data loading, refresh has a PTR spinner)
        if dataSource.isLoadingProducts && refreshState != .loading {
```

We shouldn't show the loading cell if we're refreshing – the PTR indicator is sufficient, and changing the state breaks the refresh task anyway.

Reply:
I implemented it to support the error "retry" case, but we could live without it. It's a shame there's no way to manually show a PTR loading animation. I remember trying to replicate it without success.